google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,346 @@
+    """
+    Given a gcs_prefix this funciont will return a list of documentai.Documents
+    from files that follow the pattern pages-.*-to-.*
+    """
+    base_shards = []
+    revision_shards = []
+    text = []
+    match = re.match(r"gs://(.*?)/(.*)", gcs_prefix)
+
+    if match is None:
+        raise ValueError("gcs_prefix does not match accepted format")
+
+    output_bucket, output_prefix = match.groups()
+
+    file_check = re.match(r"(.*[.].*$)", output_prefix)
+
+    if file_check is not None:
+        raise ValueError("gcs_prefix cannot contain file types")
+
+    text, base_bytes, revision_bytes = _get_base_and_revision_bytes(
+        output_bucket, output_prefix
+    )
+    page_pb = documentai.Document.Page.pb()
+    pb = documentai.Document.pb()
+    for byte in base_bytes:
+        doc = documentai.Document()
+        doc.pages = [page_pb.FromString(byte)]
+        doc.text = text
+        base_shards.append(doc)
+
+    for byte in revision_bytes:
+        revision_shards.append(pb.FromString(byte))
+
+    return base_shards, revision_shards
+
+
+def _modify_docproto(entities):
+    entities_array = []
+    history = []
+    for e in entities:
+        if (
+            e.provenance.type_ == _OP_TYPE.ADD
+            or e.provenance.type_ == _OP_TYPE.OPERATION_TYPE_UNSPECIFIED
+        ):
+            entities_array.append(Entity(documentai_entity=e))
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                }
+            )
+        elif e.provenance.type_ == _OP_TYPE.REMOVE:
+            entity = entities_array.pop(int(e.id) - 1)
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": entity.type_,
+                    "original_text": entity.mention_text,
+                }
+            )
+            del entity
+        elif e.provenance.type_ == _OP_TYPE.REPLACE:
+            history.append(
+                {
+                    "object": "Entity",
+                    "entity_provenance_type": _OP_TYPE(e.provenance.type_).name,
+                    "original_entity": e,
+                    "original_type": e.type_,
+                    "original_text": e.mention_text,
+                    "replace_type": e.type_,
+                    "replace_text": e.mention_text,
+                }
+            )
+            entities_array[int(e.id) - 1].replace(documentai_entity=e)
+    return entities_array, history
+
+
+def _get_revised_documents(revision: documentai.Document):
+    """
+    Given the immutable_document and revisions this function
```

Comment:
Can you add some more explicit definitions? I still don't really know what "provenance" means. And a lot of the API specs mark this as deprecated.

Reply:
Also, why does this function exist? It's a hidden function that only returns output from another hidden function